### ISSUE TITLE
[BUG] Helm redoc templates - flyte chart - not honoring redoc.enabled in values

### ISSUE THREAD

**Issue by cmontemuino** [flyteorg/flyte#4006]
### Describe the bug

When installing the `flyte` chart and setting `redoc.enabled` to `false`, the `redoc` deployment and service is still installed.

The problem is in the `charts/flyte/templates/redoc/deployment.yaml` and `charts/flyte/templates/redoc/deployment.yaml`. These need to be wrapped by a conditional on the `rdoc.enabled` value, like it is done with other templates (for example, `templates/redis`).

### Expected behavior

No `redoc` deployment when installing the `flyte` chart with `redoc.enabled` set to `false`.

### Additional context to reproduce

```shell
helm repo add flyteorg https://flyteorg.github.io/flyte
helm repo update
helm upgrade flyte -n flyte flyteorg/flyte \
    --set redoc.enabled=false \
    --dry-run > flyte.yaml
cat flyte.yaml| grep "name: redoc"
```

Above command should not output the `redoc` deployment and service.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by flixr** [comment]
see also https://github.com/flyteorg/flyte/pull/2982

**Comment by cmontemuino** [comment]
Hi @flixr! It seems like I've missed your previous PR, although it's not completely tackling my reported issue. Your PR is updating the service only (vs service + deployment), and it's for another Chart (flyte-deps vs flyte).

**Comment by flixr** [comment]
@eapolinario this is an easy fix, please see #5452
